tensorflow/core/kernels/training_ops.cc

```diff
@@ -328,6 +328,45 @@ struct ApplyAdamSYCL {
 template <typename T>
 struct ApplyAdam<CPUDevice, T> : ApplyAdamNonCuda<CPUDevice, T> {};
 
+template <typename Device, typename T>
+struct ApplyAdaMaxNonCuda {
+  void operator()(const Device& d, typename TTypes<T>::Flat var,
+                  typename TTypes<T>::Flat m, typename TTypes<T>::Flat v,
+                  typename TTypes<T>::ConstScalar beta1_power,
+                  typename TTypes<T>::ConstScalar beta2_power,
+                  typename TTypes<T>::ConstScalar lr,
+                  typename TTypes<T>::ConstScalar beta1,
+                  typename TTypes<T>::ConstScalar beta2,
+                  typename TTypes<T>::ConstScalar epsilon,
+                  typename TTypes<T>::ConstFlat grad, bool use_nesterov) {
+    if (use_nesterov) {
+      LOG(WARNING) << "AdaMax doesn't support use_nesterov yet, ignore it.";
```

Comment:
If it doesn't support the argument let's remove it from the interface

Reply:
(adding arguments later is fine)

Reply:
Yes, I don't like it as well. However I also want to make Adamax reuse Adam's interface mostly. I'd like to find another way to solve the problem.

Reply:
I don't think there's value in reusing the interface at this point. The interface people generally care about using is the tensorflow Optimizer interface, and you can implement that regardless of nesterov or not. So remove.

Reply:
OK. I'll work on it in the weekend.

Reply:
`use_nesterov` has been removed.